cumulusci/core/tasks.py

```diff
@@ -103,6 +115,15 @@ def _validate_options(self):
             if config.get("required") is True and name not in self.options:
                 missing_required.append(name)
 
+            elif config.get("type") and name in self.options:
+                datatype = config.get("type")
+                try:
+                    self.options[name] = datatype.convert(
+                        self.options[name], name, None
+                    )
+                except click.exceptions.UsageError as e:
+                    raise TaskOptionsError(str(e)) from e
```

Comment:
Using click's types for this is an interesting idea. I need to think through whether that covers everything we'll want to do, plays nice with options both from the CLI and from YAML, and is compatible with the possibility of starting to use Python type checking. Can we start a quip doc to reach consensus on what the design goals are?

Reply:
https://salesforce.quip.com/oHIAALXWwR0P